### PR TITLE
Java ITs: Migrate to jUnit 5

### LINKED ISSUE #3507: Replace RunWith(Suite.class) with RunWith(AllTests.class)
Currently we run the Java Integration Tests with a "test suite" in order to start SonarQube only once per the full suite of tests.

Until #3484 , the tests had to be manually added to a list (hardcoded). This means that when adding a new IT, if the IT wasn't also added to the list in the `Tests` class, the test wouldn't run. With #3484, there's an additional test that makes sure all tests are present in the `Suite` list.

The next step, for maintainability, would be to take the logic from `EnsureAllTestsRunTest` and add it directly in `Tests` by replacing `@RunWith(Suite.class)` with `@RunWith(AllTests.class)` - to take advantage of the JUnit API.  See this [SO answer](https://stackoverflow.com/questions/3257080/how-do-i-dynamically-create-a-test-suite-in-junit-4).

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Part of #4223
Fixes #3507

~~This is based on the version bump~~

First part:
* Migrate to jUnit5, use jUnit5 suite.
* Orchestrator is started by the 1st test, from all the tests that will run in series
* Orchestrator is never stopped

### CODE REVIEW

**Review by gregory-paidis-sonarsource — APPROVED** [review]
Looks good!

I like that you removed the weird `chain` methods in some test classes.
I took a look into Java UTs+ITs total counts. It looks like this:

- UTs: same as before
- C# ITs: lost  one test (expected, the `EnsureAllTestRuns` class was nuked)
- VB+Shared ITs: lost two tests (expected, same reason as before, _twice_)

Based on the above, we should consider marking #3507 as fixed by this possibly, since we do not need to answer the `did everything run?` question anymore.

I left a comment for my own clarification, you can choose how to handle it. 
Approving.

**Review by gregory-paidis-sonarsource — CHANGES_REQUESTED** [review]
Just noticed: `Shared` entrypoint is missing this:

```java

  @BeforeAll
  public static void deleteLocalCache() {
    TestUtils.deleteLocalCache();
  }
```

File: its/src/test/java/com/sonar/it/csharp/CasingAppTest.java [hunk 1]
```diff
@@ -36,21 +34,21 @@
...
-  @Before
-  public void init() {
-    TestUtils.reset(ORCHESTRATOR);
+  @TempDir
+  private static Path temp;
+
+  @BeforeAll
+  public static void init() throws IOException {
+    TestUtils.initLocal(ORCHESTRATOR);
+    Tests.analyzeProject(temp, PROJECT);
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/CasingAppTest.java:45]: No.
I've removed `initLocal` in later PR and moved the special `BeforeAll` initialization to their single-test places where possible.

This is a special feature and should be used only where it makes sense.

**Comment by pavel-mikula-sonarsource** [comment]
> ust noticed: `Shared` entrypoint is missing this:
> 
> ```java
>   @BeforeAll
>   public static void deleteLocalCache() {
>     TestUtils.deleteLocalCache();
>   }
> ```

Yep, it will get reworked in #7597 so we don't need to fix it here

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**